pallets/asset-manager/src/tests.rs

```diff
@@ -0,0 +1,730 @@
+// Copyright 2021 Parallel Finance Developer.
+// This file is part of Parallel Finance.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Tests for AssetManager Pallet
+use crate::*;
+use mock::*;
+
+use frame_support::{
+	assert_noop, assert_ok,
+	storage::migration::{put_storage_value, storage_key_iter},
+	Blake2_128Concat,
+};
+use xcm::latest::prelude::*;
+
+#[test]
+fn registering_works() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(AssetManager::register_asset(
+			Origin::root(),
+			MockAssetType::MockAsset(1),
+			0u32.into(),
+			1u32.into(),
+			true
+		));
+
+		assert_eq!(
+			AssetManager::asset_id_type(1).unwrap(),
+			MockAssetType::MockAsset(1)
+		);
+		assert_eq!(
+			AssetManager::asset_type_id(MockAssetType::MockAsset(1)).unwrap(),
+			1
+		);
+		expect_events(vec![crate::Event::AssetRegistered {
+			asset_id: 1,
+			asset: MockAssetType::MockAsset(1),
+			metadata: 0u32,
+		}])
+	});
+}
+
+#[test]
+fn test_asset_exists_error() {
+	new_test_ext().execute_with(|| {
+		assert_ok!(AssetManager::register_asset(
+			Origin::root(),
+			MockAssetType::MockAsset(1),
+			0u32.into(),
+			1u32.into(),
+			true
+		));
+
+		assert_eq!(
+			AssetManager::asset_id_type(1).unwrap(),
+			MockAssetType::MockAsset(1)
+		);
+		assert_noop!(
+			AssetManager::register_asset(
+				Origin::root(),
+				MockAssetType::MockAsset(1),
+				0u32.into(),
+				1u32.into(),
+				true
+			),
+			Error::<Test>::AssetAlreadyExists
+		);
```

Comment:
Please also cover test case of `ErrorCreatingAsset`

Reply:
test added @ashi20192020ayush